### PR TITLE
Handle sequence number wrap around

### LINKED ISSUE #29: Handle sequence number wrap around

### PR DESCRIPTION
**Pull request by kasperisager**
Closes #29

### CODE REVIEW

File: src/seq.c [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+int32_t
+udx__seq_diff (uint32_t a, uint32_t b) {
+  return a - b;
+}
+
+int
+udx__seq_compare (uint32_t a, uint32_t b) {
+  int32_t d = udx__seq_diff(a, b);
+  return d < 0 ? -1 : d > 0 ? 1 : 0;
```
Comment (kasperisager) [inline src/seq.c:21]: This might actually be flipped 🤔

File: src/udx.c [hunk 2]
```diff
@@ -407,7 +419,7 @@ process_packet (udx_t *self, char *buf, ssize_t buf_len) {
     process_sacks(stream, buf, buf_len);
   }
 
-  if (type & UDX_HEADER_DATA_OR_END && udx__cirbuf_get(inc, seq) == NULL && (stream->status & UDX_STREAM_SHOULD_READ) == UDX_STREAM_READ) {
+  if (type & UDX_HEADER_DATA_OR_END && seq_compare(stream->ack, seq) <= 0 && udx__cirbuf_get(inc, seq) == NULL && (stream->status & UDX_STREAM_SHOULD_READ) == UDX_STREAM_READ) {
```
Comment (kasperisager) [inline src/udx.c:422]: This checks if `seq` is either the next or a future packet. Is that correct?

Comment (mafintosh) [reply src/udx.c:421]: yes!

File: src/udx.c [hunk 3]
```diff
@@ -490,7 +501,9 @@ process_packet (udx_t *self, char *buf, ssize_t buf_len) {
     }
   }
 
-  while (stream->remote_acked < ack) {
+  int16_t len = seq_diff(ack, stream->remote_acked);
```
Comment (mafintosh) [inline src/udx.c:504]: int32_t ?

**Pull request merged**